omod/src/test/java/org/openmrs/web/controller/ForgotPasswordFormControllerTest.java

```diff
@@ -0,0 +1,281 @@
+		
+		for (int x = 1; x <= 5; x++) {
+			MockHttpServletRequest request = new MockHttpServletRequest();
+			request.setMethod("POST");
+			
+			request.addParameter("uname", "invaliduser");
+			
+			controller.handleRequest(request, new MockHttpServletResponse());
+		}
+		
+		// those were the first five, now the sixth request (with a valid user) should fail
+		MockHttpServletRequest request = new MockHttpServletRequest();
+		request.setMethod("POST");
+		
+		request.addParameter("uname", "validuser");
+		request.addParameter("secretAnswer", "valid secret answer");
+		controller.handleRequest(request, new MockHttpServletResponse());
+		
+		Assert.assertFalse(Context.isAuthenticated());
+	}
+	
+	/**
+	 * If a user enters 5 requests with username+secret answer, the 6th should fail even if that one
+	 * has a valid answer in it
+	 *
+	 * @throws Exception
+	 */
+	@Test
+	public void shouldLockOutAfterFiveFailedInvalidSecretAnswers() throws Exception {
+		ForgotPasswordFormController controller = (ForgotPasswordFormController) applicationContext.getBean("forgotPasswordForm");
+		
+		for (int x = 1; x <= 5; x++) {
+			MockHttpServletRequest request = new MockHttpServletRequest();
+			request.setMethod("POST");
+			
+			request.addParameter("uname", "validuser");
+			request.addParameter("secretAnswer", "invalid secret answer");
+			
+			controller.handleRequest(request, new MockHttpServletResponse());
+		}
+		
+		// those were the first five, now the sixth request (with a valid user) should fail
+		MockHttpServletRequest request = new MockHttpServletRequest();
+		request.setMethod("POST");
+		
+		request.addParameter("uname", "validuser");
+		request.addParameter("secretAnswer", "valid secret answer");
+		
+		controller.handleRequest(request, new MockHttpServletResponse());
+		
+		Assert.assertFalse(Context.isAuthenticated());
+	}
+	
+	/**
+	 * If a user enters 4 username requests, the 5th one should reset the lockout and they should be
+	 * allowed 5 attempts at the secret answer
+	 *
+	 * @throws Exception
+	 */
+	@Test
+	public void shouldGiveUserFiveSecretAnswerAttemptsAfterLessThanFiveFailedUsernameAttempts() throws Exception {
+		ForgotPasswordFormController controller = new ForgotPasswordFormController();
+		
+		for (int x = 1; x <= 4; x++) {
+			MockHttpServletRequest request = new MockHttpServletRequest();
+			request.setMethod("POST");
+			
+			request.addParameter("uname", "invaliduser");
+			
+			controller.handleRequest(request, new MockHttpServletResponse());
+		}
+		
+		// those were the first four, now the fifth is a valid username
+		MockHttpServletRequest request = new MockHttpServletRequest();
+		request.setMethod("POST");
+		
+		request.addParameter("uname", "validuser");
+		
+		controller.handleRequest(request, new MockHttpServletResponse());
+		
+		Assert.assertNotNull(request.getAttribute("secretQuestion"));
+		
+		// now the user has 5 chances at the secret answer
+		
+		// fifth request
```

Comment:
I meant to put request 5 and 6 to avoid duplication of code but it's okay if you wish to leave it this way